clustering/k-means-clustering-plugin/plugin.json

```diff
@@ -0,0 +1,130 @@
+{
+  "name": "K-Means Clustering",
+  "version": "0.3.2",
+  "title": "K-Means Clustering",
+  "description": "Cluster the data using K-Means.",
+  "author": "Jayapriya Nagarajan (jayapriya.nagarajan@nih.gov), Kelechi Nina Mezu (nina.mezu@nih.gov) Hamdah Shafqat Abbasi (hamdahshafqat.abbasi@nih.gov)",
+  "institution": "National Center for Advancing Translational Sciences, National Institutes of Health",
+  "repository": "https://github.com/polusai/polus-plugins",
+  "website": "https://ncats.nih.gov/preclinical/core/informatics",
+  "citation": "",
+  "containerId": "polusai/k-means-clustering-plugin:0.3.2",
+  "baseCommand": [
+    "python3",
+    "-m",
+    "polus.plugins.clustering.k_means"
+  ],
+  "inputs": [
+    {
+      "name": "inpDir",
+      "type": "genericData",
+      "description": "Input tabular data",
+      "required": "true"
+    },
+    {
+      "name": "filePattern",
+      "type": "string",
+      "description": "Pattern to parse input files",
+      "required": false
+    },
+    {
+      "name": "methods",
+      "type": "enum",
+      "description": "Select Manual or Elbow or Calinski Harabasz or Davies Bouldin methods",
+      "options": {
+        "values": [
+          "Manual",
+          "Elbow",
+          "CalinskiHarabasz",
+          "DaviesBouldin",
+          "default"
+        ]
+      },
+      "required": "true"
+    },
+    {
+      "name": "minimumRange",
+      "type": "number",
+      "description": "Enter minimum k-value:",
+      "required": "false"
+    },
+    {
+      "name": "maximumRange",
+      "type": "number",
+      "description": "Enter maximum k-value:",
+      "required": "false"
+    },
+    {
+      "name": "numOfClus",
+      "type": "number",
+      "description": "Number of clusters:",
+      "required": "false"
+    },
+    {
+      "name": "fileExtension",
+      "type": "enum",
+      "description": "File format of an output file",
+      "required": true,
+      "options": {
+        "values": [
+          ".csv",
+          ".feather",
+          ".parquet",
+          ".hdf5",
+          ".arrow",
+          "default"
+        ]
+      }
+    }
+  ],
+  "outputs": [
+    {
+      "name": "outDir",
+      "description": "Output collection",
+      "type": "genericData",
+      "required": true
+    }
+  ],
+  "ui": [
+    {
+      "key": "inputs.inpDir",
+      "title": "Input tabular data",
+      "description": "Input tabular data for clustering"
+    },
+    {
+      "key": "inputs.filePattern",
+      "title": "FilePattern",
+      "description": "Pattern to parse input files",
+      "default": ".arrow"
```

Comment:
The input value should be a universal regex, such as `.+`

Reply:
done